Python project names should not include '-' character

Causes a problem with the generated .NET namespace

For example, use the --project-name ```bad-name```:
```tap python new-project --project-name bad-name --directory bad-name```

Try to build using ```dotnet build```:
```
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  Restored c:\src\bad-name\bad-name.Api\bad-name.Api.csproj (in 4.15 sec).
c:\src\bad-name\bad-name.Api\bad-name.Api.csproj : warning OpenTAP Install: Installer : Package 'Python' contains possi
bly relevant plugins for next package installations, but these will not be loaded.
c:\src\bad-name\bad-name.Api\ExampleApi.cs(6,20): error CS0116: A namespace cannot directly contain members such as fie
lds, methods or statements [c:\src\bad-name\bad-name.Api\bad-name.Api.csproj]
c:\src\bad-name\bad-name.Api\ExampleApi.cs(6,14): error CS1514: { expected [c:\src\bad-name\bad-name.Api\bad-name.Api.c
sproj]
c:\src\bad-name\bad-name.Api\ExampleApi.cs(7,1): error CS1022: Type or namespace definition, or end-of-file expected [c
:\src\bad-name\bad-name.Api\bad-name.Api.csproj]

Build FAILED.

c:\src\bad-name\bad-name.Api\bad-name.Api.csproj : warning OpenTAP Install: Installer : Package 'Python' contains possi
bly relevant plugins for next package installations, but these will not be loaded.
c:\src\bad-name\bad-name.Api\ExampleApi.cs(6,20): error CS0116: A namespace cannot directly contain members such as fie
lds, methods or statements [c:\src\bad-name\bad-name.Api\bad-name.Api.csproj]
c:\src\bad-name\bad-name.Api\ExampleApi.cs(6,14): error CS1514: { expected [c:\src\bad-name\bad-name.Api\bad-name.Api.c
sproj]
c:\src\bad-name\bad-name.Api\ExampleApi.cs(7,1): error CS1022: Type or namespace definition, or end-of-file expected [c
:\src\bad-name\bad-name.Api\bad-name.Api.csproj]
    1 Warning(s)
    3 Error(s)

Time Elapsed 00:00:14.04
```

From ExampleApi.cs:
![image](https://github.com/opentap/OpenTap.Python/assets/69871845/66b60502-b035-4d21-874f-d6ccf9c1c539)

--- a/OpenTap.Python/CliActions/NewProject.cs
+++ b/OpenTap.Python/CliActions/NewProject.cs
@@ -24,13 +24,28 @@ namespace OpenTap.Python.CliActions
         [CommandLineArgument("project-name",  Description = "The name of the newly create project.")] 
         public string ProjectName { get; set; }
         
-
         public int Execute(CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(ProjectName))
+            if (string.IsNullOrWhiteSpace(ProjectName))
                 throw new ArgumentException("The project name (--project-name) must be set.", nameof(ProjectName));
             if (string.IsNullOrEmpty(Directory))
                 throw new ArgumentException("The output directory (--directory) must be set.", nameof(Directory));
+            
+            ProjectName = ProjectName.Trim();
+            { // validate project name
+                if (!(char.IsLetter(ProjectName[0]) || ProjectName[0] == '_'))
+                {
+                    throw new ArgumentException("The project name (--project-name) must start with a letter or underscore.", nameof(ProjectName));
+                }
+                foreach (var letter in ProjectName)
+                {
+                    if (!(char.IsLetterOrDigit(letter) || letter == '_'))
+                    {
+                        throw new ArgumentException("The project name (--project-name) must only include digits, letters or underscores.", nameof(ProjectName));
+                    }
+                }
+            }
+            
             
             var pythonPluginVersion = Installation.Current.FindPackage("Python")?.Version.ToString() ?? "3.1";
             

